src/common/tusb_fifo.c

```diff
@@ -39,31 +39,49 @@
 // implement mutex lock and unlock
 #if CFG_FIFO_MUTEX
 
-static void tu_fifo_lock(tu_fifo_t *f)
+static void tu_fifo_lock_wr(tu_fifo_t *f)
```

Comment:
this is minor but we don't really need 2 pair of lock/unlock for read/write. We should have `tu_fifo_lock(mutex)` and pass either f->mutex_wr or f->mutex_rd, and update the current function accordinly.

Reply:
Well yes but within tusb_fifo.c we always use `tu_fifo_lock_wr(f);` etc. in any case and if we change it to the version you suggested then we always woud need `#if #end` stuff to exclude `tu_fifo_lock(f->mutex_wr)` because `f->mutex_wr` is only defined in this case. I leave it up to you what you prefer ok? :)

Reply:
it is an macro, macro can swallow argument without any problems :smiley:

Reply:
Nice, you there is always something new ^^. Already done!